source/hp/dof_handler.cc

```diff
@@ -227,7 +227,7 @@ namespace internal
           vertex_fe_association (dof_handler.finite_elements->size(),
                                  std::vector<bool> (dof_handler.tria->n_vertices(), false));
 
-          for (typename DoFHandler<dim,spacedim>::active_cell_iterator
+          for (TriaActiveIterator < dealii::DoFCellAccessor < DoFHandler < dim, spacedim >, false > >
```

Comment:
What happens if you use `typename hp::DoFHandler<dim,spacedim>::active_cell_iterator` instead in these places? This is really what the compiler is supposed to select, and it wouldn't be overly onerous to make this change.

If that doesn't help, can you provide a backtrace of the place where it fails?